spec/services/discovery_engine/sync/locking_spec.rb

```diff
@@ -0,0 +1,54 @@
+RSpec.describe DiscoveryEngine::Sync::Locking do
+  subject(:lockable) { Class.new.include(described_class).new }
```

Comment:
I have never seen this before! That's a neat way to create a test class and include the module being tested. 😮

Reply:
Yeah, I hate how RSpec doesn't have `described_module` as an alias for `described_class`, so it sounds a bit silly - but it's pretty neat!